runtime/domain.c

```diff
@@ -630,28 +699,146 @@ CAMLexport void caml_reset_domain_lock(void)
   return;
 }
 
-void caml_init_domains(uintnat minor_heap_wsz) {
-  int i;
-  uintnat size;
-  void* heaps_base;
+/* minor heap initialization and resizing */
 
+static uintnat get_minor_heap_reservation_size() {
   /* sanity check configuration */
-  if (caml_mem_round_up_pages(Bsize_wsize(Minor_heap_max))
-          != Bsize_wsize(Minor_heap_max))
-    caml_fatal_error("Minor_heap_max misconfigured for this platform");
+  if (caml_mem_round_up_pages(Bsize_wsize(caml_minor_heap_max_wsz))
```

Comment:
This should be an assertion rather than a check that runs always.

Reply:
Actually we need to be careful here, because there is no such rounding check in `gc_ctrl.c` which can change the value. With the PR as it currently stands (an assertion instead of a runtime check), the user can pass non-rounded values in `Gc.set` and the code runs.

If I understand correctly, this check is here for performance reasons, not for correctness: we are worried about negative performance impact of two minor heap areas sharing the same page? (Or is it an artifact of an old implementation using bit tricks that we don't need anymore?)

I'm not sure whether the correct change is:
- remove the check altogether
- maintain the invariant that `caml_minor_heap_max_wsz` is always rounded up (when it is initially defined, and when it is changed)

Reply:
Calls to `mmap` must be page-aligned: "If addr is not NULL, then the kernel takes it as a hint about where to place the mapping; on Linux, the kernel will pick a nearby page boundary" (https://man7.org/linux/man-pages/man2/mmap.2.html). The same holds for committing memory via `VirtualAlloc`: "If the memory is already reserved and is being committed, the address is rounded down to the next page boundary." (https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualalloc)

I think that rounding up is the better developer experience, so I'll make the change.